icubam/backoffice/handlers/base.py

```diff
@@ -13,9 +14,25 @@ def initialize(self, config, db):
     self.config = config
     self.db = db
     self.user = None
+    self.client = tornado.httpclient.AsyncHTTPClient()
+
+  async def ping(self):
+    servers = {'server': 'www', 'messaging': 'sms'}
+    result = {}
+    for server, key in servers.items():
+      url = self.config[server].base_url + 'health'
+      try:
+        resp = await self.client.fetch(
+          tornado.httpclient.HTTPRequest(url=url, request_timeout=1))
+        result[key] = {'status': resp.code == 200, 'since': resp.body}
+      except:
+        result[key] = {'status': False, 'since': '-'}
+        continue
+    return result
 
-  def render(self, path, **kwargs):
-    super().render(path, this_user=self.user, **kwargs)
+  async def render(self, path, **kwargs):
+    pings = await self.ping() if self.user and self.user.is_admin else {}
```

Comment:
Should this be handled by a periodic callback instead? It is easy to define one in tornado it seems and will be better than pinging at each request.

Reply:
sure. let me give it a try

Reply:
Tentatively approving as discussed. You can change it later.

Reply:
done !